sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdConnectionStateListener.java

```diff
@@ -65,12 +73,33 @@ public void onException(Throwable exception) {
                 logger.debug("Will not raise the connection state change event for error", exception);
             }
         }
+
+        openConnectionIfNeeded();
     }
 
     public RntbdConnectionStateListenerMetrics getMetrics() {
         return this.metrics;
     }
 
+    private void openConnectionIfNeeded() {
+        if (this.endpoint.isClosed()) {
+            return;
+        }
+
+        // If there are still channels opened, do nothing
+        if (this.endpoint.channelsMetrics() > 0) {
```

Comment:
probably should change to 1

Reply:
I would also replace it already with a constant - maybe even an env variable. Just in case we need to open this

Reply:
Like MinActiveChannels=1